### PR TITLE
Fix dangerouslySetInnerHTML being reapplied repeatedly whenever a vnode renders

### PR DESCRIPTION
**Pull request by rpetrich**
Two adjustments had to be made to prevent the DOM subtree from unnecessary updating on a node with `dangerouslySetInnerHTML` set:
1. Only reapply dangerouslySetInnerHTML when the inner __html value changes
2. Don't diff children for vnodes that have a dangerouslySetInnerHTML attribute

### CODE REVIEW

File: src/dom/index.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ export function setAccessor(node, name, old, value, isSvg) {
 		}
 	}
 	else if (name==='dangerouslySetInnerHTML') {
-		if (value) node.innerHTML = value.__html || '';
+		if (typeof value !== "undefined") node.innerHTML = value;
```
Comment (developit) [inline src/dom/index.js]: I think `if (value!==undefined)` is a bit cheaper (gzip/runtime)

**Comment by developit** [comment]
@rpetrich - any chance the `value.__html !== old.__html` check fixes this issue on it's own? It seems like, while it might end up resulting in setAccessor being called erroneously like it is today, it would prevent re-applying `innerHTML` when the value has not changed.

**Comment by rpetrich** [comment]
@developit It doesn't. `innerHTML` creates a DOM subtree that's unmanaged by preact, which will then be destroyed by `innerDiffNode` if a check isn't added.

_(at least that's my memory of this patchset—it's been awhile since I've reviewed it in depth)_

**Comment by cristianbote** [comment]
This is addressing #844

**Comment by marvinhagemeister** [comment]
This is fixed with Preact X :tada:

**Pull request closed without merging**